zerver/lib/email_notifications.py

```diff
@@ -117,6 +118,54 @@ def make_emoji_img_elem(emoji_span_elem: CSSSelector) -> Dict[str, Any]:
     content = lxml.html.tostring(fragment).decode('utf-8')
     return content
 
+def fix_spoilers_in_html(content: str, language: str) -> str:
+    with override_language(language):
+        spoiler_title: str = _("Open Zulip to see the spoiler content")
+    fragment = lxml.html.fromstring(content)
+    spoilers = fragment.find_class("spoiler-block")
+    for spoiler in spoilers:
+        header = spoiler.find_class("spoiler-header")[0]
+        spoiler_content = spoiler.find_class("spoiler-content")[0]
+        header_content = header.find("p")
+        if header_content is None:
+            # Create a new element to append the spoiler to)
+            header_content = lxml.html.fromstring("<p></p>")
```

Comment:
`div` is a block element, and `p` is also a block element. Thus, we need this handling. Maybe look at removing this redundancy and just have `p.spoiler-header` instead?

This could have been added for better styling on the webapp but we should be able to get around it hopefully.